mkosi

```diff
@@ -557,6 +557,17 @@ def mount_image(args, workspace, loopdev, root_dev, home_dev, srv_dev, root_read
 
             umount(root)
 
+@complete_step("Assigning hostname")
+def assign_hostname(args, workspace):
+    if not args.hostname:
+        return
+
+    root = os.path.join(workspace, "root")
+    hostname_path = os.path.join(root, "etc/hostname")
+    
+    with open(hostname_path, "w+") as f:
```

Comment:
> If this is by design (@poettering ?)
Yes, this is by design.

And yupp, we should remove /etc/hostname if none is set, I am very sure. Would love to see this PR updated for that.